airflow/providers/amazon/aws/utils/waiter_with_logging.py

```diff
@@ -68,23 +70,81 @@ def wait(
         except WaiterError as error:
             if "terminal failure" in str(error):
                 raise AirflowException(f"{failure_message}: {error}")
-            status_string = _format_status_string(status_args, error.last_response)
-            log.info("%s: %s", status_message, status_string)
+            log.info("%s: %s", status_message, _LazyStatusFormatter(status_args, error.last_response))
```

Comment:
Just so I'm understanding this correctly, the `__str__` of `_LazyStatusFormatter` will only be called if the log is actually being printed, which saves computation time right?

Reply:
yes, we pass the object itself to the logger, and it only calls `__str__` if the log level requires it.